__tests__/app.test.js

```diff
@@ -0,0 +1,40 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data/index");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("/api/topics", () => {
+  test("GET:200 sends an array of topics", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.topics.length).toBe(3)
+        response.body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic).toHaveProperty("description");
+        });
+      });
+  });
+
+  test("GET:400 sends an appropriate message when given an invalid path", () => {
+    return request(app)
+      .get("/api/topicnotatopic")
+      .expect(400)
+      .then((response) => { 
+          
+        expect(response.body.msg).toBe('Bad Request')
+       
+        });
+      });
```

Comment:
Excellent, well done on implementing the custom 404 handler and testing it.

Little detail but be careful of formatting such as the random white space lines on line 35 and 37, it may be worth making sure your **Prettier** extension is switched on, so it tidies this sort of stuff up for you when you save your work.